Customer supplier invoice on dropshipping

lepistone opened the pull request:
fixes https://github.com/odoo/odoo/issues/4131 with a separate addon

lepistone committed:
initial test: so and po case

lepistone committed:
real implementation of _need_two_invoices

lepistone committed:
add test: po on delivery

lepistone committed:
view and correct context for need_two_invoices

lepistone committed:
test_get_journal_type, purchase only

lepistone committed:
test_get_journal_type, sale only

lepistone committed:
use function field instead of method in tests

lepistone committed:
first integration test, test other journal type

lepistone committed:
integration test, invoice on purchase

lepistone committed:
fix: default string was always true

lepistone committed:
finally create two invoices

lepistone committed:
use the correct partner in the generated invoices

lepistone committed:
improve the wizard view

lepistone committed:
readme

pedrobaeza commented on customer_supplier_invoice_on_dropshipping/README.rst:
```diff
@@ -0,0 +1,24 @@
+Contributors
+------------
+
+* Leonardo Pistone <leonardo.pistone@camptocamp.com>
```
You should use OCA template in https://github.com/OCA/maintainers-tools/blob/master/template/module/README.rst

pedrobaeza commented:
Thanks for this very needed feature!

I see as a better name  _stock_dropshipping_dual_invoice_, to follow conventions and short it. What do you think?

Please check Travis error also.

lepistone commented:
Thanks as usual for reviewing @pedrobaeza!

The red build is due to `stock_route_transit` changing the default dropshipping route. Before just isolating the two modules, I'll take a moment to discuss about that with @gurneyalex.

For the title I agree `dropshipping_dual_invoice` is clearer and shorter, thanks! However I wonder: do we really need to repeat `stock_` everywhere? This module, in fact, interacts with sales, purchases, stock and accounting. Moreover, all those modules are already grouped in a repository dealing with stock.

pedrobaeza commented:
Yeah, I agree that we can omit stock_ part, but I put it because the module that starts all of this is called _stock_dropshipping_, and this way, all related modules are seen together in all lists (module list in Odoo, repository list, and so on).

lepistone committed:
rename module to stock_dropshipping_dual_invoice

lepistone committed:
isolate stock_route_transit

This module modifies the standard drop shipping route, and other tests
might depend on it. Specifically, those of
stock_dropshipping_dual_invoice.

lepistone committed:
adapt readme to OCA template

pedrobaeza commented:
Thanks!

:+1:

lepistone committed:
link sale to out invoice, purchase to in invoice

Otherwise, they are linked to both, which does not make sense.

lepistone committed:
do not show the two invoices together, doc

lepistone committed:
show the route field on the SOL

gurneyalex commented:
:+1:
